docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+
+The format of `trippie.txt` is as follows:
+```
+DEFAULT <trip_index>
+<trip_index - 1>,<trip_name>,<trip_date (dd-mm-yyyy)>,<trip_max_day>
+1,Amazing Bali,03-04-2021,3
+2,Australia,01-01-2021,8
+3,Wonderful Singapore,02-01-2021,9
+```
+Here the `trip_name` serves both as the Trip's name and the file name, hence it should not contain invalid characters for file names, i.e. `<>:"/\|?*.` A sample directory structure containing 3 trips is as the following:
+```
+trippie_data
+    ├── Amazing Bali.txt
+    ├── Australia.txt
+    ├── Wonderful Singapore.txt
+    └── trippie.txt
+```
+
+
+
+### 4.2 Sorting Place List (Kian En)
+
+The way Trippie sorts its place list, regardless of the order the places are added by the user, is through **bubble sort**. The sorting algorithm is called every time a new place is added by the user. 
+
+When listing either places, Trippie will sort them based on both *Day* and *Time*. Hence, there is a need to use a **stable** sorting algorithm. Bubble sort is used for its simplicity and easy implementation. Furthermore, since this sorting algorithm is called every time a new place is added, the list will always be sorted, allowing the time complexity of the sorting algorithm to be **O(n)**.
+
+Likewise when sorting expenses, Trippie will sort them based on *Day*.
+
+### 4.3 Budget and Expenses (Shawn)
+
+Trippie provides travellers an easy and convenient way to track expenses while constantly ensuring that the budget is not exceeded.
+
+<kbd>
+
+![Expenses UML Diagram](https://i.imgur.com/xuZNMxG.jpg)
+
+<center><i>Figure 7: Expenses Classes</i></center>
+</kbd>
+
+The `NewTripCommand` receives `currencyAbbreviation`, `budgetValue` and `ForExValue` from the inputs and stores it within the current `Trip` object. `ExpenseList` array object which consists of `Expense` object uitilized the relevant objects stored in `Trip`. This allows `AddExpenseCommand`, `DeleteExpenseCommand` and `ListExpenseCommand` to access the data and edit or list them respectively.
+`AddExpenseCommand` consist of `ExpenseComparator` which implements `Comparator<Expense>`. The `ExpenseComparator` sorts the variables in a blackbox by <b>comparing</b> two objects at a time.
+`ListExpenseCommand` further access the `currencyAbbreviation` and `ForExValue` objects to print out both local and foreign currency based on the input provided.
+
+A sample output can be seen as follows:
+```
+Total budget: $300.00 MYR (200.00 SGD)
+Expense List:
+[1] Day 1: bubble tea - $8.00
+[2] Day 2: hotel room - $200.00
+Your current total spending is $208.00 MYR (138.67 SGD)
+Your current remaining budget is $92.00 MYR (61.33 SGD)
+You are still spending within your budget.
+[=======---] 69.3%
+```
+
+### 4.4 Foreign Exchange Converter (Shawn)
+
+Trippie provide travellers with a quicker alternative to convert any amount into local or foreign currency.
+
+<kbd>
+
+![Currency UML Diagram](https://i.imgur.com/wxM71AO.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/49357313/97660738-6c165780-1aad-11eb-92ae-550e5e293d14.png)

I think that if a variable is represented with association and role there is no need to include them in the class. For example the currencyChoice and currencyAmount.